tests/functional_tests/ffront_tests/test_execution.py

```diff
@@ -906,3 +907,48 @@ def consume_constants(input: Field[[IDim], np.float32]) -> Field[[IDim], np.floa
     output = np_as_located_field(IDim)(np.zeros((1,), dtype=np.float32))
     consume_constants(input, out=output, offset_provider={})
     assert np.allclose(np.asarray(output), constants.PI * constants.E)
+
+
+def test_scalar_scan():
+    size = 10
+    KDim = Dimension("K", kind=DimensionKind.VERTICAL)
+    qc = np_as_located_field(IDim, KDim)(np.zeros((size, size)))
+    scalar = 1.0
+    expected = np.full((size, size), np.arange(start=1, stop=11, step=1).astype(float64))
+
+    @scan_operator(axis=KDim, forward=True, init=(0.0))
+    def _scan_scalar(carry: float, qc_in: float, scalar: float) -> float:
+        qc = qc_in + carry + scalar
+        return qc
+
+    @program
+    def scan_scalar(qc: Field[[IDim, KDim], float], scalar: float):
+        _scan_scalar(qc, scalar, out=qc)
+
+    scan_scalar(qc, scalar, offset_provider={})
+    assert np.allclose(np.asarray(qc), expected)
+
+
+def test_tuple_scalar_scan():
+    size = 10
+    KDim = Dimension("K", kind=DimensionKind.VERTICAL)
+    qc = np_as_located_field(IDim, KDim)(np.zeros((size, size)))
+    qc2 = np_as_located_field(IDim, KDim)(np.zeros((size, size)))
+    scalar = 1.0
+    expected = np.full((size, size), np.arange(start=1, stop=11, step=1).astype(float64))
+
+    @scan_operator(axis=KDim, forward=True, init=(0.0, 0.0))
+    def _scan_tuple_scalar(
+        state: tuple[float, float], qc_in: float, scalar: float
+    ) -> tuple[float, float]:
+        return (qc_in + state[0] + scalar, qc_in + state[1] - scalar)
+
+    @field_operator
+    def scan_tuple_scalar(
+        qc: Field[[IDim, KDim], float], scalar: float
+    ) -> tuple[Field[[IDim, KDim], float], Field[[IDim, KDim], float]]:
+        return _scan_tuple_scalar(qc, scalar)
+
+    scan_tuple_scalar(qc, scalar, out=(qc, qc2), offset_provider={})
+    assert np.allclose(np.asarray(qc), expected)
+    assert np.allclose(np.asarray(qc2), -expected)
```

Comment:
I meant a tuple argument for the scan operator (since this is what you are implementing here), something like (didn't fix expected).

```suggestion
def test_tuple_scalar_scan():
    size = 10
    KDim = Dimension("K", kind=DimensionKind.VERTICAL)
    qc = np_as_located_field(IDim, KDim)(np.zeros((size, size)))
    scalar = 1.0
    expected = np.full((size, size), np.arange(start=1, stop=11, step=1).astype(float64))

    @scan_operator(axis=KDim, forward=True, init=(0.0))
    def _scan_scalar(carry: float, qc_in: float, scalar: tuple[float, float]) -> float:
        qc = qc_in + carry + scalar[0] + scalar[1]
        return qc

    @program
    def scan_scalar(qc: Field[[IDim, KDim], float], scalar: tuple[float, float]):
        _scan_scalar(qc, scalar, out=qc)

    scan_scalar(qc, (scalar, scalar), offset_provider={})
    assert np.allclose(np.asarray(qc), expected)

```

Reply:
I propose to delete the `is_field_type_or_tuple_of_field_type(type_: ts.TypeSpec)` and replace it by a more generic version
```python
def is_type_or_tuple_of_type(
    type_: ts.TypeSpec, expected_type: ts.TypeSpec | tuple[ts.TypeSpec, ...]
) -> bool:
    if not isinstance(expected_type, tuple):
        expected_type = (expected_type,)
    return all(isinstance(t, expected_type) for t in primitive_constituents(type_))
```